### PR TITLE
rename docs/ --> doc/

### PR DESCRIPTION
**Pull request by akaszynski**
### Documentation directory rename

Our documentation directory has been named ``docs/`` since the creation of this project, and it's not inline with ``scipy``, ``numpy``, ``pandas``, ``scikit-learn``, ``matplotlib``, and many others that use simply ``doc``.

Should be a simple fix to rename the directory.  Any objections?

### CODE REVIEW

**Comment by adeak** [comment]
Works for me. I keep trying to tab-complete from `cd doc`, and the `docker*` directories keep interfering.

**Comment by MatthewFlamm** [comment]
It is easy enough to understand the current structure, and there are many places it is referenced.  But, it shouldn't matter if you catch all the places that use it.  You are missing several places where `docs` is used.  I found these using `git grep docs`.  Some of these have multiple lines that need to be fixed.

I cannot comment on lines not touched unfortunately, so I'm listing them here instead:

- CONTRIBUTING.md
- MANIFEST.in
- docker_dev/build.sh
- docs/conf.py  <- `"doc_path": "docs"`, I'm not sure if this one needs to be changed
- docs/extras/developer_notes.rst
- setup.cfg

**Comment by MatthewFlamm** [comment]
Found some more:

- `README.rst:.. image:: https://github.com/pyvista/pyvista/raw/master/docs/_static/pyvista_banner_small.png`
- `tests/plotting/test_plotting.py:See the image regression notes in docs/extras/developer_notes.rst`

File: README.rst [hunk 1]
```diff
@@ -2,8 +2,8 @@
 PyVista
 #######
 
-.. image:: https://github.com/pyvista/pyvista/raw/master/docs/_static/pyvista_banner_small.png
-   :target: https://docs.pyvista.org/examples/index.html
+.. image:: https://github.com/pyvista/pyvista/raw/master/doc/_static/pyvista_banner_small.png
+   :target: https://doc.pyvista.org/examples/index.html
```
Comment (MatthewFlamm) [inline README.rst:6]: Is the url going to change?  If so, this seems like a bigger change than I thought.

Comment (akaszynski) [reply README.rst:6]: URL won't change (good thing too).

**Comment by akaszynski** [comment]
> tests/plotting/test_plotting.py:See the image regression notes in docs/extras/developer_notes.rst

Done.  Ready for review (again).

**Comment by akaszynski** [comment]
I'd like to have this merged so we can avoid any merge conflicts as this is a fairly big change.

**Comment by adeak** [comment]
I'll take a look soon.

File: .gitignore [hunk 2]
```diff
@@ -54,13 +54,20 @@ htmlcov
...
+<<<<<<< HEAD
+api-autogen/
+_autosummary/
+doc/external_examples.rst
+
+=======
 _autosummary
 
 docs/getting-started/external_examples.rst
+>>>>>>> main
```
Comment (adeak) [inline .gitignore:70]: Merge fail!

File: azure-pipelines.yml [hunk 3]
```diff
@@ -243,22 +243,28 @@ jobs:
...
   - script: |
+<<<<<<< HEAD
+      make -C doc html || true  # will fail on VTK9
+      make -C doc html -b coverage SPHINXOPTS="-W"
+      cat doc/_build/coverage/python.txt
+=======
       make -C docs phtml || true  # will fail on VTK9
       make -C docs phtml -b coverage SPHINXOPTS="-W"
       cat docs/_build/coverage/python.txt
+>>>>>>> main
```
Comment (adeak) [inline azure-pipelines.yml:259]: Merge fail!

File: setup.cfg [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [build_sphinx]
 source-dir = docs
```
Comment (adeak) [inline setup.cfg:2]: Should this be `doc` too?
```suggestion
source-dir = doc
```

**Comment by MatthewFlamm** [comment]
It looks good from just looking at the changes, but I am still trying to figure out how to build the docs locally.

File: .gitignore (hunk 2 above)
Comment (akaszynski) [reply .gitignore:70]: Gah!

File: azure-pipelines.yml [hunk 5]
```diff
@@ -243,22 +243,22 @@ jobs:
...
+      path: '$(Build.SourcesDirectory)/doc/examples'
     displayName: Build pyvista doc examples cache
 
   # run this twice since VTK9 fails to close render windows
   - script: |
-      make -C docs phtml || true  # will fail on VTK9
-      make -C docs phtml -b coverage SPHINXOPTS="-W"
-      cat docs/_build/coverage/python.txt
+      make -C doc html || true  # will fail on VTK9
+      make -C doc html -b coverage SPHINXOPTS="-W"
```
Comment (adeak) [inline azure-pipelines.yml:252]: Can't we keep `phtml` for speed?
```suggestion
      make -C doc phtml || true  # will fail on VTK9
      make -C doc phtml -b coverage SPHINXOPTS="-W"
```

**Review by adeak — APPROVED** [review]
Approving with one suggestion (based on one of the pre-merge versions).

File: azure-pipelines.yml (hunk 5 above)
Comment (akaszynski) [reply azure-pipelines.yml:252]: Sorry, yes, we can.

**Pull request merged**